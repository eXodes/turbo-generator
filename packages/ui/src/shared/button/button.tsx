"use client";

import { JSX, ReactNode } from "react";

interface ButtonProps {
  children: ReactNode;
  className?: string;
  appName: string;
}

const Button = ({ appName, className, children }: ButtonProps): JSX.Element => {
  return (
    <button
      className={className}
      onClick={() => alert(`Hello from your ${appName} app!`)}
    >
      {children}
    </button>
  );
};

export { type ButtonProps, Button };
