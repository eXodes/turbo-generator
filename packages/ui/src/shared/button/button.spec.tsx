import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";

import { Button } from "./button";

describe("Button", () => {
  it("should have 'Button' page title", async () => {
    const { baseElement } = render(<Button appName={"App"}>Button</Button>);

    expect(baseElement).toMatchSnapshot();
  });
});
