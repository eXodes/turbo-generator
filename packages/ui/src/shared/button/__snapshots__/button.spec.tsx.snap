// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should have 'Button' page title 1`] = `
<body>
  <div>
    <button>
      Button
    </button>
  </div>
</body>
`;
